common/worker_versioning/worker_versioning.go

```diff
@@ -41,14 +42,43 @@ import (
 )
 
 const (
-	buildIdSearchAttributePrefixAssigned    = "assigned"
-	buildIdSearchAttributePrefixVersioned   = "versioned"
-	buildIdSearchAttributePrefixUnversioned = "unversioned"
-	BuildIdSearchAttributeDelimiter         = ":"
+	buildIdSearchAttributePrefixReachability = "reachability"
+	buildIdSearchAttributePrefixAssigned     = "assigned"
+	buildIdSearchAttributePrefixVersioned    = "versioned"
+	buildIdSearchAttributePrefixUnversioned  = "unversioned"
+	BuildIdSearchAttributeDelimiter          = ":"
 	// UnversionedSearchAttribute is the sentinel value used to mark all unversioned workflows
 	UnversionedSearchAttribute = buildIdSearchAttributePrefixUnversioned
 )
 
+// escapeBuildIdSearchAttributeDelimiter is a helper which escapes the BuildIdSearchAttributeDelimiter character in the input string
+func escapeBuildIdSearchAttributeDelimiter(s string) string {
+	s = strings.Replace(s, BuildIdSearchAttributeDelimiter, `|`+BuildIdSearchAttributeDelimiter, -1)
+	return s
+}
+
+// ReachabilityBuildIdSearchAttribute returns the search attribute value for the currently assigned build ID in the form
+// 'reachability:<behavior>:<deployment_series_name>:<deployment_build_id>'
+func ReachabilityBuildIdSearchAttribute(behavior enumspb.VersioningBehavior, deployment *deploymentpb.Deployment) string {
+	var escapedDeployment string
+	if deployment == nil {
+		escapedDeployment = "UNVERSIONED"
```

Comment:
hmm... behavior is not meaningful in this case, do we need to add a `reachability:..` value in this case? (an `unversioned` values is already added in BuildIds)

Reply:
yeah honestly I think I never call this with a nil deployment, I was just not thinking / being paranoid about nil pointer error

Reply:
I can remove the nil-check

Reply:
interesting, in the general case, it is possible that the behavior is unpinned but deployment is nil (unpinned override of a new/unversioned wf).

So this is another evidence that we should not make it generic, and keep it specific to pinned when the deployment cannot be null.